packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -224,10 +240,17 @@ class Plugin implements CLNPlugin {
             await _handlingSubscription(
                 event: jsonRequest.method, request: param);
           } else {
-            var result = await _call(jsonRequest.method, param);
-            var response =
-                Response(id: jsonRequest.id, result: result).toJson();
-            stdout.write(json.encode(response));
+            if (!rpcMethods.containsKey(jsonRequest.method)) {
+              var result = await _callHook(jsonRequest.method, param);
+              var response =
+                  Response(id: jsonRequest.id, result: result).toJson();
+              stdout.write(json.encode(response));
+            } else {
+              var result = await _call(jsonRequest.method, param);
+              var response =
+                  Response(id: jsonRequest.id, result: result).toJson();
+              stdout.write(json.encode(response));
+            }
```

Comment:
can you remove this code and move the call inside the `_rpc_call_method`

Somethink like that

```dart
  Future<Map<String, Object>> _call(
      String name, Map<String, Object> request) async {
    if (rpcMethods.containsKey(name)) {
      var method = rpcMethods[name]!;
      return await method.call(this, request);
    } else {
       return await _callHook(name, request);
    }
   throw Exception("Method with name $name not found!");
  }
```

with this call our can restore the old code in the start method, should be more simple in this way also read the code

Reply:
Fixed.